### ISSUE TITLE
Example crashes with Vulkan backend

### ISSUE THREAD

**Issue by arlohb** [gfx-rs/wgpu#3218]
**Description**
When using the Vulkan backend, it crashes immediately after opening. The error doesn't occur with the GL backend instead.

**Repro steps**
Cloning the repository, and running the hello-triangle example.

**Expected vs observed behavior**
It was meant to run normally, showing a triangle on the screen. Instead, the window quickly closed after it opened and multiple vulkan errors were shown.

**Output**

[2022-11-18T12:01:39Z ERROR wgpu_hal::vulkan::instance] GENERAL [../mesa-22.2.3/src/vulkan/runtime/vk_semaphore.c:148 (0x0)]
    	Combination of external handle types is unsupported for VkSemaphore creation. (VK_ERROR_INVALID_EXTERNAL_HANDLE)
[2022-11-18T12:01:39Z ERROR wgpu_hal::vulkan::instance] 	objects: (type: DEVICE, hndl: 0x558fc88ccb00, name: ?)
[2022-11-18T12:01:39Z ERROR wgpu::backend::direct] Error in Adapter::request_device: not enough memory left
thread 'main' panicked at 'Failed to create device: RequestDeviceError', wgpu/examples/hello-triangle/main.rs:35:10

I'm not sure what other logs you want, I'm happy to add more information to help.

**Platform**
Arch Linux, AMD Radeon integrated graphics.
This is happening on the v0.13, v0.14, and master branch, these are the only ones I've tested

**Comment by allen-marshall** [comment]
I'm having the same problem on the master branch (haven't tested others yet). One noteworthy thing is that I can see an empty window appear for a split second before the program crashes. My output from `cargo run --example hello-triangle` is as follows; it's about the same as what arlohb encountered.

```
    Finished dev [unoptimized + debuginfo] target(s) in 0.13s
     Running `target/debug/examples/hello-triangle`
[2022-11-23T13:10:29Z ERROR wgpu_hal::vulkan::instance] GENERAL [../mesa-22.2.3/src/vulkan/runtime/vk_semaphore.c:148 (0x0)]
    	Combination of external handle types is unsupported for VkSemaphore creation. (VK_ERROR_INVALID_EXTERNAL_HANDLE)
[2022-11-23T13:10:29Z ERROR wgpu_hal::vulkan::instance] 	objects: (type: DEVICE, hndl: 0x55cdfbffb170, name: ?)
[2022-11-23T13:10:29Z ERROR wgpu::backend::direct] Error in Adapter::request_device: not enough memory left
thread 'main' panicked at 'Failed to create device: RequestDeviceError', wgpu/examples/hello-triangle/main.rs:35:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I am on Arch Linux with dual Radeon HD 8970M cards. I tried switching between the amdgpu and radeon drivers and got the same issue in both cases. I noticed that the error message is slightly similar to the error from #3201, which involves AMD hardware. I wonder if something AMD-specific is causing this issue too. I plan to do more testing and at least see if I can identify a specific commit where the issue starts.

**Comment by allen-marshall** [comment]
After looking at #3201 some more, I found that switching my Vulkan driver from AMDVLK to RADV seems to resolve my issue, just as it did in #3201. I also found that `vulkaninfo` segfaults on my machine unless I switch to RADV, which is a pretty strong indicator that the problem lies outside `wgpu`, at least in my case. I found out that, in terms of official support, AMDVLK stopped supporting my hardware in version 2021.Q2.6, so I think I will just use RADV going forward.

I tried out past versions of `wgpu`, and it seems like the hello-triangle example crashes all the way back to when it was first introduced, which makes sense if the problem is external to `wgpu`.
